### DISCUSSION TITLE
Asynchronous Rendering

### DISCUSSION THREAD

**Discussion by billwanjohi** [starship/starship#4532] [Ideas]
I got very excited when spaceship-prompt [announced asynchronous rendering](https://spaceship-prompt.sh/blog/2022-spaceship-v4/), only to later remember that I had actually left it for the faster starship.

This new non-blocking behavior (and the faster time-to-first-keystroke it allows) is more important to me than raw speed, so my eyes are wandering back to that older project.

I noticed that async was mentioned in a roadmap discussion back in 2020, but are there any more current plans to introduce this possibility?

**Comment by davidkna** [comment]
I don't think anyone is working on asynchronous rendering.

On the other hand, much of the groundwork for external command cache has been implemented, which could offer some of the same advantages - but I don't think anyone is working on that either at the moment. For git, gitoxide could improve the general performance of things like git status once the required features are implemented.

**Reply by syphar** [reply]
also there are some shell-specific ways to achieve this, like [`fish-async-prompt`](https://github.com/acomagu/fish-async-prompt)

**Reply by syphar** [reply]
~@davidkna is there an issue with more info about the command cache?~ found it: #3863
